### PR TITLE
added activity module

### PR DESCRIPTION
**Pull request by ghost**
This module implements the activity model. The model will be used in payroll and timesheet modules to identify what an employee did in a worked day or a timesheet record (he worked on a job position or took leave).

### CODE REVIEW

File: hr_activity/__openerp__.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+{
+    'name': 'Activity',
+    'category': 'Localization',
+    'version': '1.0',
+    'license': 'AGPL-3',
+    'category': 'Generic Modules/Human Resources',
```
Comment (max3903) [inline hr_activity/__openerp__.py:25]: category is twice

File: hr_activity/__openerp__.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+Contributors
+------------
+* David Dufresne <david.dufresne@savoirfairelinux.com>
+* Pierre Lamarche <pierre.lamarche@savoirfairelinux.com>
+""",
+    'author': 'Savoir-faire Linux',
+    'website': 'https://www.savoirfairelinux.com/',
+    'depends': [
+        'hr',
+        'hr_holidays',
```
Comment (max3903) [inline hr_activity/__openerp__.py:52]: No need for hr. hr_holidays already depends on hr.

File: hr_activity/__openerp__.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    ],
+    'data': [
+        'security/ir.model.access.csv',
+        'hr_holiday_status_view.xml',
+        'hr_holiday_status_data.xml',
+        'hr_activity_view.xml',
+        'hr_activity_data.xml',
```
Comment (max3903) [inline hr_activity/__openerp__.py]: put data files in data/ and view files in view/

File: hr_activity/hr_activity_view.xml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        </record>
+
+        <record id="open_tree_activity" model="ir.actions.act_window">
+            <field name="name">Activities</field>
+            <field name="res_model">hr.activity</field>
+            <field name="view_type">form</field>
+            <field name="search_view_id" ref="view_hr_activity_filter"/>
+        </record>
+
+        <menuitem action="open_tree_activity" id="menu_hr_activity_tree" parent="hr.menu_hr_configuration" sequence="12"/>
```
Comment (max3903) [inline hr_activity/hr_activity_view.xml]: Can you put one attribute per line ?

Comment (ghost) [inline hr_activity/hr_activity_view.xml]: Done!

----- Mail original -----
De: "max3903" notifications@github.com
À: "OCA/hr" hr@noreply.github.com
Cc: "David Dufresne" david.dufresne@savoirfairelinux.com
Envoyé: Mercredi 12 Novembre 2014 09:21:24
Objet: Re: [hr] added activity module (#36)

In hr_activity/hr_activity_view.xml: > +            <field name="model">hr.activity</field>

> -            <field name="arch" type="xml">
> -                <search string="Activity">
> -                    <field name="name" string="Timesheet Activity"/>
> -                </search>
> -             </field>
> -        </record>
>   +
> -        <record id="open_tree_activity" model="ir.actions.act_window">
> -            <field name="name">Activities</field>
> -            <field name="res_model">hr.activity</field>
> -            <field name="view_type">form</field>
> -            <field name="search_view_id" ref="view_hr_activity_filter"/>
> -        </record>
>   +
> -        <menuitem action="open_tree_activity" id="menu_hr_activity_tree" parent="hr.menu_hr_configuration" sequence="12"/> 

Can you put one attribute per line ?

File: hr_activity/__openerp__.py (hunk 3 above)
Comment (LeartS) [inline hr_activity/__openerp__.py]: Is there a convention on this? I see a lot of OCA modules that have view data in the "root" directory.

Comment (ghost) [inline hr_activity/__openerp__.py]: Do you think it is overkill?

I think it improves readability when there is many files in a folder.

----- Mail original -----
De: "Leonardo Donelli" notifications@github.com
À: "OCA/hr" hr@noreply.github.com
Cc: "David Dufresne" david.dufresne@savoirfairelinux.com
Envoyé: Mercredi 12 Novembre 2014 09:43:23
Objet: Re: [hr] added activity module (#36)

In hr_activity/**openerp**.py: > +------------

> +\* David Dufresne david.dufresne@savoirfairelinux.com
> +\* Pierre Lamarche pierre.lamarche@savoirfairelinux.com
> +""",
> -    'author': 'Savoir-faire Linux',
> -    'website': 'https://www.savoirfairelinux.com/',
> -    'depends': [
> -        'hr',
> -        'hr_holidays',
> -    ],
> -    'data': [
> -        'security/ir.model.access.csv',
> -        'hr_holiday_status_view.xml',
> -        'hr_holiday_status_data.xml',
> -        'hr_activity_view.xml',
> -        'hr_activity_data.xml', 

Is there a convention on this? I see a lot of OCA modules that have view data in the "root" directory.

Comment (LeartS) [inline hr_activity/__openerp__.py]: I do not think it's overkill, I kind of prefer it, but OCA conventions and uniformity are more important than personal preferences. That's why I was aking if there is one.

P.S: try to not reply via mail or at least delete the quoted part, it's cluttering up the discussion page.

Comment (ghost) [inline hr_activity/__openerp__.py]: Hi Leonardo,
sorry for my last reply and thanks for your comment.
I asked Maxime Chambreuil and he says there is still no status taken in regard to view and data files location.

File: hr_activity/__openerp__.py [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+Activity
+========
+This module does not implement any function but only a relational model that
+can be inherited in different specific modules.
+
+Adds the "Activity" model. An activity is whether a job or a leave type.
+It is meant to be used in both timesheet and payroll modules but for different
+reasons.
+    - Timesheet: Used when one employee may have different jobs (or task) in
+        the same company witch imply different cost prices. In his timesheet,
```
Comment (pedrobaeza) [inline hr_activity/__openerp__.py]: s/witch/which

Comment (pedrobaeza) [inline hr_activity/__openerp__.py]: I can understand this concept: one employee has a fixed salary, and thus, a cost price. Why do you need to diferentiate between jobs done to determine his cost?

File: hr_activity/__openerp__.py [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+        he can enter the time he spent for every job or leave type.
+
+    - Payroll: Used in worked days to identify if a worked days line is
+        a normal worked day, a vacation day, a sick leave...
+
+Adds the vacation leave type.
+
+Adds the paid_leave boolean field on leave types (used on payslips)
```
Comment (pedrobaeza) [inline hr_activity/__openerp__.py]: This field is not related to this module, so maybe it can be put in another module that fits better.

**Comment by pedrobaeza** [comment]
In general, I see this module not too coherent and the feeling that can be omitted because it's an auxiliar structure for later computation, or splitted in the corresponding modules that uses it.

If I don't find more arguments for its inclussion, I would say :-1:

File: hr_activity/__openerp__.py (hunk 6 above)
Comment (ghost) [inline hr_activity/__openerp__.py]: I agree on this. The field fits better in another module.

File: hr_activity/__openerp__.py (hunk 5 above)
Comment (ghost) [inline hr_activity/__openerp__.py]: Because an employee is paid at different hourly rates whether he works on a job position or another, we need to differentiate how much time he spent on each job. An example is an employee in a restaurant. The same person can work as barman, waiter and busboy in the same working schedule and be remunarated differently according to each job.

Comment (ghost) [inline hr_activity/__openerp__.py]: Another example is specific to Quebec, there is a contribution of the employer called CNT. For its computation, it is required to know how many hours an employee spent on a job on which he supplied heavy equipment (like trucks, tractors, loaders...) at his own cost.

Comment (plamarche) [inline hr_activity/__openerp__.py]: Hi Pedro,
I supervise the work of David on this R&D project.

The point is that an employee can have several jobs with a specific hourly rate related to each job. (Job 1 = 10 $ / hour, Job 2 = 15 $ / hour, etc...)
The best example is in the building industry : a worker can do electrician work, plumbing work, etc... When he fills his timesheet, we need him to specifiy how many hours he worked as an electrician, etc...
As a matter of fact, we can know his salary base for the payslip period by doing the calculation by job \* specific job hourly rate.
As Odoo is using analytic accounts for the timesheets, with the architecture we have chosen with David, an employee can specify a project (analytic account) on his time sheet and then select an specific activity (job) for which he want to enter hours.
ONce the timesheet has been validated, we can import the hours from the timesheet to the Payslip preparation, with the job detailed hours to make the real calculation of the payroll possible.

Please feel free to ask if you have any further Question, Pedro.

Pierre

**Comment by ghost** [comment]
@pedrobaeza Hi Pedro, I created a new pull request to replace this one. It should me much more coherent since I merged the module with a corresponding module that uses it.
https://github.com/OCA/hr/pull/48
